app/controllers/comments_controller.rb

```diff
@@ -3,15 +3,29 @@ class CommentsController < ApplicationController
 
   def create
     @post = Post.find(params[:post_id]) #finding post
-    @comment = Comment.create(params[:comment].permit(:content)) #finding the comment with the params
-    @comment.user_id = current_user.id #assigning the user_id to current user
-    @comment.post_id = @post.id #assigning the post_id from comments to the id from posts table
+    @comment = Comment.find_or_initialize_by(post: @post, user: current_user)
+    # @comment.user_id = current_user.id #assigning the user_id to current user
+    # @comment.post_id = @post.id #assigning the post_id from comments to the id from posts table
 
-    if @comment.save
-      redirect_to post_path(@post)
+    if @comment.new_record?
+      @comment.attributes = comment_params
+      if @comment.save
+        redirect_to post_path(@post),
+        notice: "Successfully added your comment"
+      else
+        render 'new'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.